core/templates/services/state-interaction-stats.service.ts

```diff
@@ -34,34 +32,29 @@ import { UrlInterpolationService } from
   'domain/utilities/url-interpolation.service';
 
 export interface IAnswerData {
-  /* eslint-disable camelcase */
-  answer; // Type depends on interaction id.
-  frequency: number;
-  is_addressed?: boolean;
-  /* eslint-enable camelcase */
+  // Needs to be 'any' because type depends on the interaction's id.
+  'answer': any;
```

Comment:
@nishantwrp Can you help here?

Reply:
Yeah, i think we can define it's type using `|` because there are a fixed number of different cases possible. I think it's type would be similar to
https://github.com/oppia/oppia/blob/66597a96525e24fe4f01f2ffe9c1a5d929c67dca/core/templates/domain/exploration/RuleObjectFactory.ts#L37-L39
i'm not sure though.

Reply:
I think we'd be able to do this using a "discriminated union" (https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions), but the change would be very big so I'd prefer to do it in a separate PR to keep this one focused on the caching.

Reply:
Hi @brianrodri, isn't the type for this would be the same as `IRuleInput`? (I also used `discriminated union` there.)

Reply:
That isn't exactly a discriminated union, a real union would have an additional "tag" type to determine exactly which type is currently selected.

```typescript
interface IAnswer = {
  id: 'string';
  answer: string;
} | {
  id: 'fraction';
  answer: IFractionBackendDict;
} | {
  id: 'note';
  answer: INote;
} // etc...
```

However, like I said, implementing that is a rather big change and I'd prefer keeping it out of this PR.